docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1061 @@
+the Response object to the execute function in the EditCommand object.
+
+**Step 8.** The printExpenditureEdited method under the UI class is then called in the execute function and an
+acknowledgement message is printed to the user.
+
+##### Overall sequence diagram for editing an Entry
+
+![img_1.png](static/frontend/commands/EditCommandSequenceDiagram.png)
+
+#### View Command
+##### Class diagram of view command 
+![ViewCommandClassDiagram.png](./static/viewCommandClassDiagram.png)
+Class diagram above shows the methods called in the execute method of the 
+View Command object.
+
+##### Implementation
+![ViewCommandSequenceDiagram.png](static/frontend/commands/ViewCommandOverallSequenceDiagram.png)
+
+**Step 1.** String of user arguments is fed into the parseViewCommand method in Parser object.
+
+**Step 2.** The various filters such as price and dateTime are extracted from the argument String and fed into the 
+            constructor of viewCommand.
+
+**Step 3.** A request object is instantiated, with all the query parameters (eg price) as its attributes.
+
+**Step 4.** The request is fed into the backend method requestEndpointEntries, which then sends this request to the
+            EntriesEndpoint class.
+
+**Step 5.** A response object is then returned from EntriesEndpoint. This response contains the serialised form of all 
+            relevant entries.
+
+**Step 6.** The entries are then deserialised and fed into the ui printEntriesToBeViewed method, which prints the 
+            details of each entry to the user.
+
+#### Help Command
+
+##### Implementation
+
+**Step 1.** The Parser extracts the help command from the user input and calls the parseHelpCommand method.
+
+**Step 2.** The execute method of the returned HelpCommand object is then called.
+
+**Step 3.** In the execute method, the printHelp method of the UI Class is called and a fixed string displaying all 
+            available commands is called.
+
+
+#### Exit/Bye Command
+
+##### Implementation
+
+**Step 1.** The Parser extracts the bye command from the user input and calls the parseHelpCommand method.
+
+**Step 2.** An ExitCommand is returned and since this command is the only command object with the isExit attribute set
+            as "true", the while loop of the program will terminate since !isExit is the loop condition.
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+<!-- @@author -->
+
+<!-- ### UI
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div> -->
+
+<!-- @@author jinxuan-owyong -->
+## Backend
+
+The backend uses a simplified RESTful API approach. This allows us to decouple code using the proven industry practices. The following diagram illustrates the relationship between various classes involved in `Backend` as described in the [application architecture](#architecture)
+
+![Backend](./static/backend/BackendClassDiagram.png)
```

Comment:
Labels between classes like "read/write" and "handle request" should include the direction indicated by the black triangles.
<img width="283" alt="Screenshot 2023-03-30 at 13 42 16" src="https://user-images.githubusercontent.com/79389472/228740608-1c1cc46e-4ad5-484a-9120-aa2ea94a3563.png">